### PR TITLE
Fixes #5453 updates cffi & argon2_cffi

### LINKED ISSUE #5453: cffi and argon2_cffi needs update for buiding in Ubuntu 20.04
## Description

The current values fail to build on `Ubuntu Focal 20.04`.

```
  wrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DUSE__THREAD -DHAVE_SYNC_SYNCHRONIZE -I/usr/include/ffi -I/usr/include/libffi -I/usr/include/python3.8 -c c/_cffi_backend.c -o build/temp.linux-x86_64-3.8/c/_cffi_backend.o", "    c/_cffi_backend.c: In function ‘b_do_dlopen’:", "    c/_cffi_backend.c:4197:31: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]", "     4197 |         *p_printable_filename = PyText_AsUTF8(s);", "          |                               ^", "    c/_cffi_backend.c: In function ‘b_callback’:", "    c/_cffi_backend.c:5911:5: warning: ‘ffi_prep_closure’ is deprecated: use ffi_prep_closure_loc instead [-Wdeprecated-declarations]", "     5911 |     if (ffi_prep_closure(closure, &cif_descr->cif,", "          |     ^~", "    In file included from c/_cffi_backend.c:15:", "    /usr/include/x86_64-linux-gnu/ffi.h:334:1: note: declared here", "      334 | ffi_prep_closure (ffi_closure*,", "          | ^~~~~~~~~~~~~~~~", "    In file included from c/cffi1_module.c:20,", "                     from c/_cffi_backend.c:7370:", "    c/call_python.c: In function ‘_get_interpstate_dict’:", "    c/call_python.c:20:30: error: dereferencing pointer to incomplete type ‘PyInterpreterState’ {aka ‘struct _is’}", "       20 |     builtins = tstate->interp->builtins;", "          |                              ^~", "    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1", "
 
```

The following two packages should be upgraded:

- cffi == 1.14.0
- argon2_cffi == 20.1.0

This is part of #4768 for the package build on Focal

### PR DESCRIPTION
**Pull request by kushaldas**
## Status

Ready for review 

## Description of Changes

Fixes #5453

Updates both the releases to the latest, these versions builds
properly on Python3.8 on Ubuntu Focal 20.04

## Testing

- [x] CI is green
- [x] Verify hashes introduced in the PR (for the source tar.gz files with the following)
- [x] https://github.com/freedomofpress/securedrop-debian-packaging/wiki/cffi-1.11.5-to-1.14.2
- [x] https://github.com/freedomofpress/securedrop-debian-packaging/wiki/argon2_cffi-18.1.0-to-20.1.0

## Deployment

Any special considerations for deployment? Consider both:

1. Upgrading existing production instances.
2. New installs.

### If you made changes to the server application code:

- [ ] Linting (`make lint`) and tests (`make test`) pass in the development container

### If you made changes to `securedrop-admin`:

- [ ] Linting and tests (`make -C admin test`) pass in the admin development container

### If you made changes to the system configuration:

- [ ] [Configuration tests](https://docs.securedrop.org/en/latest/development/testing_configuration_tests.html) pass

### If you made non-trivial code changes:

- [ ] I have written a test plan and validated it for this PR

### If you made changes to documentation:

- [ ] Doc linting (`make docs-lint`) passed locally

### If you added or updated a code dependency:

Choose one of the following:

- [x] I have performed a diff review and pasted the contents to [the packaging wiki](https://github.com/freedomofpress/securedrop-debian-packaging/wiki)
- [ ] I would like someone else to do the diff review

### CODE REVIEW

File: securedrop/requirements/python3/securedrop-app-code-requirements.txt [hunk 1]
```diff
@@ -38,40 +31,35 @@ babel==2.5.1 \
...
-    --hash=sha256:ca1bd81f40adc59011f58159e4aa6445fc585a32bb8ac9badf7a2c1aa23822f2 \
-    --hash=sha256:d5d8555d9bfc3f02385c1c37e9f998e2011f0db4f90e250e5bc0c0a85a813085 \
-    --hash=sha256:e55e22ac0a30023426564b1059b035973ec82186ddddbac867078435801c7801 \
-    --hash=sha256:e90f17980e6ab0f3c2f3730e56d1fe9bcba1891eeea58966e89d352492cc74f4 \
-    --hash=sha256:ecbb7b01409e9b782df5ded849c178a0aa7c906cf8c5a67368047daab282b184 \
-    --hash=sha256:ed01918d545a38998bfa5902c7c00e0fee90e957ce036a4000a88e3fe2264917 \
-    --hash=sha256:edabd457cd23a02965166026fd9bfd196f4324fe6032e866d0f3bd0301cd486f \
-    --hash=sha256:fdf1c1dc5bafc32bc5d08b054f94d659422b05aba244d6be4ddc1c72d9aa70fb \
-    # via argon2-cffi, cryptography
+cffi==1.14.0 \
```
Comment (emkll) [inline securedrop/requirements/python3/securedrop-app-code-requirements.txt]: cffi version reviewed in the PR description is 1.14.2, as such, the hashes are different. Should we use 1.14.2 here?

Comment (kushaldas) [reply securedrop/requirements/python3/securedrop-app-code-requirements.txt]: I will update it in some time.

**Review by emkll — COMMENTED** [review]
Went through the test plan and everything looks good to me: 

- [x] CI is green
- [x] Verify hashes introduced in the PR (for the source tar.gz files with the following)
- [x] https://github.com/freedomofpress/securedrop-debian-packaging/wiki/cffi-1.11.5-to-1.14.2
- [x] https://github.com/freedomofpress/securedrop-debian-packaging/wiki/argon2_cffi-18.1.0-to-20.1.0

One last question for you @kushaldas : would it make sense to track the same version of cffi (1.4.2) in develop-requirements.txt as well? It's currently on 1.4.0 [1](which may have been the source of the initial confusion). It seems fine for me to merge as-is, but curious what you think.

[1] https://github.com/freedomofpress/securedrop/blob/upgrade_ffi_twins/securedrop/requirements/python3/develop-requirements.txt#L109

**Review by emkll — APPROVED** [review]
Thanks @kushaldas , good to merge when CI passes

**Pull request merged**